plugins/zenoh-plugin-storage-manager/src/lib.rs

```diff
@@ -307,20 +262,37 @@ impl From<StorageRuntimeInner> for StorageRuntime {
     }
 }
 
+impl PluginControl for StorageRuntime {
+    fn report(&self) -> PluginReport {
+        PluginReport::default()
+    }
+    fn plugins_status(&self, names: &keyexpr) -> Vec<PluginStatusRec> {
+        let guard = self.0.lock().unwrap();
+        guard
+            .plugins_manager
+            .plugins_status(names)
+            .into_iter()
+            .map(PluginStatusRec::into_owned)
+            .collect()
+    }
+}
+
 impl RunningPluginTrait for StorageRuntime {
-    fn config_checker(&self) -> ValidationFunction {
+    fn config_checker(
+        &self,
+        _: &str,
+        old: &serde_json::Map<String, serde_json::Value>,
+        new: &serde_json::Map<String, serde_json::Value>,
+    ) -> ZResult<Option<serde_json::Map<String, serde_json::Value>>> {
         let name = { zlock!(self.0).name.clone() };
-        let runtime = self.0.clone();
-        Arc::new(move |_path, old, new| {
-            let old = PluginConfig::try_from((&name, old))?;
-            let new = PluginConfig::try_from((&name, new))?;
-            log::info!("old: {:?}", &old);
-            log::info!("new: {:?}", &new);
-            let diffs = ConfigDiff::diffs(old, new);
-            log::info!("diff: {:?}", &diffs);
-            { zlock!(runtime).update(diffs) }?;
-            Ok(None)
-        })
+        let old = PluginConfig::try_from((&name, old))?;
+        let new = PluginConfig::try_from((&name, new))?;
+        log::info!("old: {:?}", &old);
```

Comment:
Is the log intended to stay there or used during development for testing purposes?
If it is meant to stay then I would provide a richer log message. As well, I'd move to the log to debug level.

Reply:
This came from existing code without changes. But I agree, this is better to put into debug level. Fixed.